src/tags.psm1

```diff
@@ -0,0 +1,1374 @@
+
+.EXAMPLE
+$taggingRule = '{"Description": null,"Enabled": true,"RuleConditionGroup": {"LogicalJoinType": "And","Children": [{"TaggingGroupingCondition": {"ObjectAttribute": "Name","CompareType": "Contains","CompareValue": ".demo"},"TaggingGroupingConditionGroup": null}]}}' | ConvertFrom-Json
+Test-SafeguardAssetTaggingRule -AssetPartitionId 1 -Tag prod -TaggingRule $taggingRule
+#>
+function Test-SafeguardAssetTaggingRule {
+    [CmdletBinding()]
+    Param(
+        [Parameter(Mandatory=$false)]
+        [string]$Appliance,
+        [Parameter(Mandatory=$false)]
+        [object]$AccessToken,
+        [Parameter(Mandatory=$false)]
+        [switch]$Insecure,
+        [Parameter(Mandatory=$false)]
+        [object]$AssetPartition,
+        [Parameter(Mandatory=$false)]
+        [int]$AssetPartitionId = $null,
+        [Parameter(Mandatory=$true,Position=0)]
+        [object]$Tag,
+        [Parameter(Mandatory=$true,Position=1)]
+        [PSCustomObject]$TaggingRule
+    )
+
+    if (-not $PSBoundParameters.ContainsKey("ErrorAction")) { $ErrorActionPreference = "Stop" }
+    if (-not $PSBoundParameters.ContainsKey("Verbose")) { $VerbosePreference = $PSCmdlet.GetVariableValue("VerbosePreference") }
+
+    # use the helper function Resolve-AssetPartitionId to get the asset partition id.
+    $AssetPartitionId = Resolve-AssetPartitionId -Appliance $Appliance -AccessToken $AccessToken -Insecure:$Insecure -AssetPartition $AssetPartition -AssetPartitionId $AssetPartitionId
+    $local:TagId = (Resolve-SafeguardTagId -AccessToken $AccessToken -Appliance $Appliance -Insecure:$Insecure -AssetPartition $AssetPartition -AssetPartitionId $AssetPartitionId $Tag)
+    if ($local:TagId) {
+        $local:RelPath = "AssetPartitions/$AssetPartitionId/Tags/$local:TagId/TestAssetRule"
+
+        Invoke-SafeguardMethod -AccessToken $AccessToken -Appliance $Appliance -Insecure:$Insecure Core POST "$local:RelPath" -Body $TaggingRule
+    } else {
+        throw "The tag with id $TagId could not be found."
+    }
+}
+
+
+
+<#
+.SYNOPSIS
+Test an AssetAccountTaggingRule on a tag.
+
+.DESCRIPTION
+Test what an AssetAccountTaggingRule would do. 
+This can be used to verify which accounts would be assigned this the tag if the rule is set.
+
+.PARAMETER Appliance
+IP address or hostname of a Safeguard appliance.
+
+.PARAMETER AccessToken
+A string containing the bearer token to be used with Safeguard Web API.
+
+.PARAMETER Insecure
+Ignore verification of Safeguard appliance SSL certificate.
+
+.PARAMETER AssetPartition
+An integer containing an ID or a string containing the name of the asset partition where the tag exists.
+
+.PARAMETER AssetPartitionId
+An integer containing the asset partition ID where the tag exists.
+(If specified, this will override the AssetPartition parameter)
+
+.PARAMETER Tag
+An integer containing the ID of the asset or a string containing the name of the tag.
+
+.PARAMETER TaggingRule
+A PSCustomObject with the JSON for the asset tagging rule.
```

Comment:
"...for the asset tagging rule." Let's do, "...for the asset account tagging rule." To be more specific and match the method name.